File: Algorithms/Python/Reverse-Integer.py
class Solution(object):

    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        ans = 0
        flag = 1 if x >= 0 else -1
        x = abs(x)
        while x != 0:
            ans = 10 * ans + x % 10
            x //= 10
        ans = flag * ans
        return 0 if ans >= 2147483647 or ans < -2147483648 else ans

File: Algorithms/Java/Reverse Integer.java
public class Solution {
    public int reverse(int x) {
    	long key = (long) x;
		long ans = 0;
		int flag = 1;
		String s;
		if (key < 0) {
			key = -key;
			flag = -1;
		}
		s = String.valueOf(key);
		int[] res = new int[s.length()];
		for (int i = 0; i <= s.length() - 1; i++) {
			res[i] = s.charAt(s.length() - 1 - i) - '0';
		}
		for (int i = res.length - 1; i >= 0; i--) {
			ans += res[res.length - 1 - i] * Math.pow(10, i);
		}
		ans *= flag;
		if (ans < -2147483648 || ans > 2147483647) {
			return 0;
		} else {
			return (int) ans;
		}
        
    }
}